From a98646feda4d16391b35f0638a6515a68b668cc5 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Sun, 20 Feb 2022 16:27:51 +0200
Subject: [PATCH] [SW-75742] gaudi: security: keep version ro

Keep version range as read only, regardless of sram access state.
This solves an issue where version info read access is denied when
rst-state message is processed by the device.

Change-Id: Ic9933b65cc8bb4388924aa31d6b3c367798fae4f
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 2f135bd..ace3328 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -2336,8 +2336,9 @@
 
 	for (i = SRAM_BGN_TO_VERS; i <= IMG_END_TO_SRAM_END; i++) {
 		cfg = &preboot_sec_dyn_cfg[i];
-		min = en ? 0 : cfg->min;
-		max = en ? 0 : cfg->max;
+		/* Keep VERSION as RO always */
+		min = (en || i == VERSION) ? 0 : cfg->min;
+		max = (en || i == VERSION) ? 0 : cfg->max;
 
 		if (cfg->rr >= 0 && cfg->min && cfg->max) {
 			preboot_secure_pcie_rr_wr(cfg->rr, cfg->min, cfg->max);
